controllers/schedule_week.go

```diff
@@ -0,0 +1,409 @@
+package controllers
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"net/http"
+	"net/url"
+	"sort"
+	"strconv"
+	"time"
+
+	"github.com/UniversityRadioYork/2016-site/models"
+	"github.com/UniversityRadioYork/2016-site/structs"
+	"github.com/UniversityRadioYork/2016-site/utils"
+	"github.com/UniversityRadioYork/myradio-go"
+	"github.com/gorilla/mux"
+)
+
+// weekFromVars extracts the year, and week strings from vars.
+func weekFromVars(vars map[string]string) (string, string, error) {
+	y, ok := vars["year"]
+	if !ok {
+		return "", "", errors.New("no year provided")
+	}
+	w, ok := vars["week"]
+	if !ok {
+		return "", "", errors.New("no week provided")
+	}
+
+	return y, w, nil
+}
+
+// weekdayFromVars extracts the year, week, and weekday strings from vars.
+func weekdayFromVars(vars map[string]string) (string, string, string, error) {
+	y, ok := vars["year"]
+	if !ok {
+		return "", "", "", errors.New("no year provided")
+	}
+	w, ok := vars["week"]
+	if !ok {
+		return "", "", "", errors.New("no week provided")
+	}
+	d, ok := vars["weekday"]
+	if !ok {
+		return "", "", "", errors.New("no weekday provided")
+	}
+
+	return y, w, d, nil
+}
+
+//
+// Week schedule algorithm
+// TODO(CaptainHayashi): move?
```

Comment:
It's difficult to tell where most of this should be.  There's also some code in `structs/schedule.go` that probably belongs in the same place as this